Mu2eG4/src/Mu2eG4Inputs.cc

```diff
@@ -88,6 +89,44 @@ namespace mu2e {
 
     }
       break; // SimParticles
+
+    case Mu2eG4PrimaryType::StageParticles: {
+      std::optional<art::ProductID> simid;
+
+      auto const h = evt.getValidHandle<StageParticleCollection>(primaryTag_);
+      for(const auto& s : *h) {
+        if(!simid) {
+          simid = s.parent().id();
+
+          // We have to read the pointed-to product into memory by hand
+          // for the get(ProductID, &handle) call below to work.
+          // See Kyle's e-mail on the art-users list on 2021-03-10.
+          *s.parent();
+        }
+        else {
+          if(*simid != s.parent().id()) {
+            throw cet::exception("BADINPUT")
+              <<"Mu2eG4Inputs::inputSimParticles(): inconsistent SimParticleCollection product ID "
+              <<"in input StageParticleCollection "
+              <<primaryTag_
+              <<std::endl;
+          }
+        }
+      }
+
+      if(simid) { // nothing to retrieve if primary inputs are emtpy
+        evt.get(*simid, res);
+        if(!res.isValid()) {
+          throw cet::exception("BADINPUT")
+            <<"Mu2eG4Inputs::inputSimParticles(): could not get SimParticleCollection "
+            <<"pointed to by StageParticles in "
+            <<primaryTag_
+            <<std::endl;
+        }
+      }
+    }
+      break; // StageParticles
```

Comment:
I am missing how this code controls for >1 stopped muons in an event (proton).  Even though that is rare, it can happen.  Which muon will be propaated in that case?  Shouldn't it be chosen randomly?

Reply:
The current code find the "last" stopped muon.  It does make sense to pick a random one instead.  Stopped muon search code is another part that will move to support multiple generators. I will add the randomization when doing the refactoring; let's leave this as is for the current PR.